code/game/objects/items/pet_carrier.dm

```diff
@@ -0,0 +1,194 @@
+#define pet_carrier_full(carrier) carrier.occupants.len >= carrier.max_occupants || carrier.occupant_weight >= carrier.max_occupant_weight
+
+//Used to transport little animals without having to drag them across the station.
+//Comes with a handy lock to prevent them from running off.
+/obj/item/pet_carrier
+	name = "pet carrier"
+	desc = "A big white-and-blue pet carrier. Good for carrying <s>meat to the chef</s> cute animals around."
+	icon = 'icons/obj/pet_carrier.dmi'
+	icon_state = "pet_carrier_open"
+	item_state = "pet_carrier"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	force = 5
+	attack_verb = list("bashed", "carried")
+	w_class = WEIGHT_CLASS_BULKY
+	throw_speed = 2
+	throw_range = 3
+	materials = list(MAT_METAL = 7500, MAT_GLASS = 100)
+	var/open = TRUE
+	var/locked = FALSE
+	var/list/occupants = list()
+	var/occupant_weight = 0
+	var/max_occupants = 3 //Hard-cap so you can't have infinite mice or something in one carrier
+	var/max_occupant_weight = MOB_SIZE_SMALL //This is calculated from the mob sizes of occupants
+
+/obj/item/pet_carrier/Destroy()
+	if(occupants.len)
+		for(var/V in occupants)
+			remove_occupant(V)
+	return ..()
+
+/obj/item/pet_carrier/Exited(atom/movable/occupant)
+	if(occupant in occupants && isliving(occupant))
+		var/mob/living/L = occupant
+		occupants -= occupant
+		occupant_weight -= L.mob_size
+
+/obj/item/pet_carrier/handle_atom_del(atom/A)
+	if(A in occupants && isliving(A))
+		var/mob/living/L = A
+		occupants -= L
+		occupant_weight -= L.mob_size
+	..()
+
+/obj/item/pet_carrier/examine(mob/user)
+	..()
+	if(occupants.len)
+		for(var/V in occupants)
+			var/mob/living/L = V
+			to_chat(user, "<span class='notice'>It has [L] inside.</span>")
+	else
+		to_chat(user, "<span class='notice'>It has nothing inside.</span>")
+	if(user.canUseTopic(src))
+		to_chat(user, "<span class='notice'>Activate it in your hand to [open ? "close" : "open"] its door.</span>")
+		if(!open)
+			to_chat(user, "<span class='notice'>Alt-click to [locked ? "unlock" : "lock"] its door.</span>")
+
+/obj/item/pet_carrier/attack_self(mob/living/user)
+	if(open)
+		to_chat(user, "<span class='notice'>You close [src]'s door.</span>")
+		playsound(user, 'sound/effects/bin_close.ogg', 50, TRUE)
+		open = FALSE
+	else
+		if(locked)
+			to_chat(user, "<span class='warning'>[src] is locked!</span>")
+			return
+		to_chat(user, "<span class='notice'>You open [src]'s door.</span>")
+		playsound(user, 'sound/effects/bin_open.ogg', 50, TRUE)
+		open = TRUE
+	update_icon()
+
+/obj/item/pet_carrier/AltClick(mob/living/user)
```

Comment:
Ghost alt click

Reply:
`if (open || !user.canUseTopic(src, be_close=TRUE))`

Reply:
oh wait this got merged